routes/artifacts.js

```diff
@@ -190,6 +190,18 @@ api.declare({
         details.bucket  = this.privateBucket.bucket;
       }
       details.prefix    = [taskId, runId, name].join('/');
+      if (input.contentMD5 || input.contentSHA256) {
+        // Remove this check, when contentMD5 and contentSHA256 are required
+        if (!input.contentMD5 || !input.contentSHA256) {
+          return res.status(400).json({
+            message:  "Both contentMD5 and contentSHA256 must be given " +
+                      "if either is given"
+          });
```

Comment:
It would be useful to include in the error message which hash was given (md5 or sha256) and which was not, and also other useful instance data for pinning down problem when troubleshooting (taskId, runId, name).

For example, the error message could be `You tried to upload artifact <name> from taskId <taskId> in run <runId>. You supplied a <hashSupplied> hash, but not a <hashNotSupplied> hash. You need to supply either both or neither.`